core/cat/factory/custom_auth_handler.py

```diff
@@ -93,37 +93,53 @@ async def authorize_user_from_jwt(
         return None
 
     async def authorize_user_from_key(
-        self,
-        user_id: str,
-        api_key: str,
-        auth_resource: AuthResource,
-        auth_permission: AuthPermission,
+            self,
+            user_id: str,
+            api_key: str,
+            auth_resource: AuthResource,
+            auth_permission: AuthPermission,
     ) -> AuthUserInfo | None:
-        http_api_key = get_env("CCAT_API_KEY")
-        ws_api_key = get_env("CCAT_API_KEY_WS")
+        http_key = get_env("CCAT_API_KEY")
+        ws_key = get_env("CCAT_API_KEY_WS")
 
-        # TODOAUTH: should we consider the user_id or just give
-        #    admin permissions to all users with the right api keys?
-
-        # chatting over websocket
-        if auth_resource == AuthResource.CONVERSATION and api_key == ws_api_key:
+        if not http_key and not ws_key:
             return AuthUserInfo(
                 id=user_id,
                 name=user_id,
                 permissions=get_base_permissions()
             )
 
-        # any http endpoint
-        if api_key == http_api_key:
+        if auth_resource == AuthResource.CONVERSATION:
+            return self._authorize_websocket_key(user_id, api_key, ws_key)
```

Comment:
There are also endpoints for conversation not related to websocket :(

Reply:
Damn, you're right! So, currently, you have to pass the WS api key for the http message endpoint if only the WS key is set? And when only the http key is set if you not passing the api key you can message freely because None == None...